docs/source/cpp/compute.rst

```diff
@@ -563,30 +563,32 @@ representation based on the rounding criterion.
 +-------------------+------------+-------------+-------------------------+----------------------------------+--------+
 | floor             | Unary      | Numeric     | Float32/Float64/Decimal |                                  |        |
```

Comment:
I would have expected that "floor" kernel is a small wrapper around "round" with a specific RoundOptions value. If the output type of "round" changes, that doesn't also change "floor"?

Reply:
Not really. Because I didn't add floor kernels for integer types, floor(int) would still be dispatched to floor(float) kernels, and thus calling the round functions for floats.

Reply:
But now that Round itself supports rounding integers, how hard would it be to expand the floor/trunc registration to integer types as well? 
(like `MakeUnaryRoundFunction` was updated to loop through all NumericTypes instead of just float32/float64)

Reply:
Is it at all useful to expose those functions for integer inputs?

Reply:
Just to have consistent behaviour with the generic round (i.e. always preserve the input type). But that alone is maybe not worth it.

Reply:
Although it might also simplify things, given that the `RoundIntegerToFloatingPointFunction` to explicitly cast int to float which is still used for floor/trunc/ceil could then be removed (but didn't check the code in detail)

Reply:
https://github.com/apache/arrow/issues/36786 I've created another issue for this.